### PR TITLE
RE: Fix title and footer text within homepage shelves

### LINKED ISSUE #10373: Fix `title` and `footer text` within homepage shelves
Per #9996, the `title` and `footer text` fields from the [shelfmodule](https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/shelfmodule/) are not limited to a number of chars. The screenshot below displays both fields at 100 characters:

114384010-dbcd5a00-9b96-11eb-80e9-4cc2c8417a3d

The maximum text length that the UI supports will also need to be determined.

### PR DESCRIPTION
**Pull request by xlisachan**
Fixes #10373 

This pull request fixes the overlap between the header and footer_text within the homepage shelves.

@bobsilverberg , I updated this with a flexbox as you had suggested, instead of using `max-width` in my initial attempts. I think this takes care of it, but, just in case, here's my thought process.

`Card-footer-` only appeared in three stylesheets:
1. `AddonsCard`
2. `Card`
3. `ShowMoreCard`

The rule that causes the footer link to appear as if it is in the Card header is located in the `AddonsCard` stylesheet. Therefore, I updated the style for `.Card-footer-link` and `.Card-footer-text` to `display: none`.

Within the stylesheet for the `Card` component, I made the following updates:
- Updated `Card-header` to become a flexbox
- Added classname of `Card-text-wrapper`, which has the same properties as `Card-footer-link` and `Card-footer-text`, and adheres to the following breakpoint rules:
  - `mediumOnly` => `display: none`
  - `large` => `text-align: right`

No changes to `ShowMoreCard`'s stylesheet was made, as none of the updates should affect this component.

While running tests, it showed that the `Addon` component was also affected, as it was displaying "Read 1 review" within the header. I updated its stylesheet to have this `display: none`, but the test still received the length of 2. I tried to console log to check that it's counting the two `div`s within the `header` tag, but could not figure it out. How do I go about checking this test? Would the length really be two even if `display` is ` none`?

Please see the screenshots of a homepage shelf below at the two breakpoints.
mediumOnly breakpoint:
Screen Shot 2021-05-11 at 8 00 52 PM

large breakpoint:
Screen Shot 2021-05-11 at 8 00 18 PM

Please review when you have a moment. Thank you!

### CODE REVIEW

File: src/amo/components/Card/index.js [hunk 1]
```diff
@@ -63,7 +63,13 @@ export default class Card extends React.Component {
           'Card--no-footer': !footer,
         })}
       >
-        {header ? <header className="Card-header">{header}</header> : null}
+        {header ? (
+          <header className="Card-header">
+            <div className="Card-header-text">{header}</div>
+
+            {footer ? <div className="Card-text-wrapper">{footer}</div> : null}
```
Comment (bobsilverberg) [inline src/amo/components/Card/index.js:70]: This `className` isn't really descriptive of what this is. Maybe something like `Card-footer-in-header` would be better?

**Review by bobsilverberg — CHANGES_REQUESTED** [review]
It's great that you got this to work, but I'm not sure how practical it is. I know I suggested it, but I have concerns. There are also some resulting layout issues that would need to be addressed. For example, on the review list page:

Before:

Screen Shot 2021-05-12 at 2 42 12 PM

After:

Screen Shot 2021-05-12 at 2 44 55 PM

I wonder if, because the only use of `Card` that really needs to be fixed is the one used for the shelves, it would be better to create a special version of `Card` just for that purpose. Then all of the style changes being made could apply to just that one card, and it shouldn't affect any other uses of `Card`. 

What do you think of that idea?

**Comment by bobsilverberg** [comment]
@xlisachan Also, I forgot to add this to my review comment, but great job getting this to work! It looks really good, but I just think there are some issues with the implementation.

**Comment by xlisachan** [comment]
Thanks for the feedback, @bobsilverberg !

Just wanted to check that we are on the same track - 
As the `HomepageShelves` component currently returns `LandingAddonsCard`s, which returns `AddonsCard`s, which returns `CardList`s, which finally returns `Card` components, the specialized `Card` component would essentially be returned by the `HomepageShelves` component?

Please let me know when you have a moment. Thanks!

**Comment by bobsilverberg** [comment]
> Thanks for the feedback, @bobsilverberg !
> 
> Just wanted to check that we are on the same track -
> As the `HomepageShelves` component currently returns `LandingAddonsCard`s, which returns `AddonsCard`s, which returns `CardList`s, which finally returns `Card` components, the specialized `Card` component would essentially be returned by the `HomepageShelves` component?
> 
> Please let me know when you have a moment. Thanks!

Just wanted to update this to say that we discussed this in our bi-weekly meeting and @xlisachan is continuing to work on this.

**Comment by xlisachan** [comment]
It's not ready for your review, but I wanted to push the commits I've made and provide an update on my progress with this fix. 

As HomepageShelves returned LandingAddonCards => AddonsCard => CardList => Card, each commit I've pushed removes a component, which hopefully makes it easier to follow when reviewing.

I am currently working on fixing the classNames for the special version of `Card` (which I've named `HomepageCard`) and `HomepageShelves`, as well as updating `TestHomepageShelves.js` and creating `TestHomepageCard.js`.

I'll let you know if I run into any issues!

**Comment by xlisachan** [comment]
The latest push includes the special version of `Card` (named `HomepageCard`), its stylesheet, and updated `HomepageShelves` component that now returns `HomepageCard`s.

As discussed, I'll work on updating the tests if we proceed with this update. I think this fix has a lot of repetitive code though and passing a props (`isHomepageCard`) through for `Card` to utilize would be a lot cleaner (see #10666)

Please see screenshots below:
Homepage (width: 1024px)- 
Screen Shot 2021-06-09 at 5 06 33 PM

Homepage (width: 768px)-
Screen Shot 2021-06-09 at 5 06 39 PM

Homepage (width: 425px)- 
Screen Shot 2021-06-09 at 5 06 46 PM

Rating  (included as it utilizes `Card` component) - 
Screen Shot 2021-06-09 at 5 07 05 PM

Review (included as it utilizes `Card` component) -
Screen Shot 2021-06-09 at 5 07 11 PM

Please review when you have a moment. Thank you!

**Comment by bobsilverberg** [comment]
@xlisachan I'm going to close this one as we decided to go the prop-drilling route.

**Pull request closed without merging**